### PR TITLE
PPL-154: Console errors and no redirect after payment completed

### PR DESCRIPTION
**Pull request by jmbeltran07**
JIRA
----
https://paystand.atlassian.net/browse/PPL-154

Description
----
With this PR we are changing the following:
- **Remove checkout.js**: The logic to open checkout changed from having an infinite observer watching whether or not the payment method was selected, to having a button rendered that triggers opening checkout. So the logic is now on paystandmagento-directpost.js
- **Include agreement section**: Agreement section is a Magento 2 core feature that we needed to make our plugin compatible with. Now, when there are terms and conditions available, Paystand's payment method will show them and inherit the requirements to check them or not before proceeding to the actual payment. **Not having this compatibility caused the order not being created (and not being redirected to thank you page) even though payment was successful.**
- **Clean console typeError logs**: This error was caused by the automatic init of checkout with previous flow, now that we are not loading checkout automatically, this error is gone
- **Checkout close button**: Now that we can open checkout 
- **Not injecting js when PS payment method is not enabled**: as pointed by customers, even though the Payment method is not enabled from configuration, we could see the errors in console due to the plugin watching and loading checkout beforehand, since this behavior is gone, now we can successfully disable paystand without impact on magento checkout
 
QA Notes
----
- Go to magento.paystand.biz (it already has this version of Plugin)
- Select your products, go to checkout and test paystand payment method 
- After paying with paystand, you should be redirected to the thank you page and an order should be created.
- After a minute or two, your order should be gone from "pending" to "processing" status, meaning the webhook was processed successfully.

### CODE REVIEW

**Comment by jmbeltran07** [comment]
Ready for code review

File: view/frontend/web/js/view/payment/method-renderer/paystandmagento-directpost.js [hunk 1]
```diff
@@ -1,15 +1,151 @@
...
+var api_domain = 'api.paystand.com';
+var checkout_domain = 'checkout.paystand.com';
+var use_sandbox = window.checkoutConfig.payment.paystandmagento.use_sandbox;
+if (use_sandbox == '1') {
+    checkoutjs_module = 'paystand-sandbox';
+    core_domain = 'paystand.co';
+    api_domain = 'api.paystand.co';
+    checkout_domain = 'checkout.paystand.co';
+    // core_domain = 'localhost:3001';
+    // api_domain = 'localhost:3001/api';
```
Comment (elliotweaver) [inline view/frontend/web/js/view/payment/method-renderer/paystandmagento-directpost.js:12]: remove localhost from code

File: view/frontend/web/js/view/payment/method-renderer/paystandmagento-directpost.js [hunk 2]
```diff
@@ -1,15 +1,151 @@
...
+        // Validate agreement section using core Magento 2 validator
+        let validateAgreementSection = function () {
+            if (agreementValidator.validate()) {
+                // if we clear agreement section, click actual ps-button to open checkout
+                $(".ps-button").click();
+            }
+        }
+
+        psCheckout.onComplete(function (data) {
+            console.log("custom checkout complete:", data);
```
Comment (elliotweaver) [inline view/frontend/web/js/view/payment/method-renderer/paystandmagento-directpost.js:121]: I would remove these two console logs. here and in onError

File: view/frontend/web/js/view/payment/method-renderer/paystandmagento-directpost.js [hunk 3]
```diff
@@ -1,15 +1,151 @@
...
+            $(".submit-trigger").click();
+        });
+
+        psCheckout.onError(function (data) {
+            console.log("custom checkout error:", data);
+        });
+
+        // uncomment for development
+        // psCheckout.checkoutEvent = function (data) {
```
Comment (elliotweaver) [inline view/frontend/web/js/view/payment/method-renderer/paystandmagento-directpost.js:130]: Remove this block of comments

File: view/frontend/web/js/view/payment/method-renderer/paystandmagento-directpost.js [hunk 4]
```diff
@@ -1,15 +1,151 @@
...
+            if (checkoutData.billing.postcode) {
+                config.payerAddressPostal = checkoutData.billing.postcode;
+            }
+            if (checkoutData.billing.regionCode) {
+                config.payerAddressState = checkoutData.billing.regionCode;
+            }
+
+            console.log("rebooting checkout with config", config);
+
+            //This block fixes the issue where checkout opens blank
```
Comment (elliotweaver) [inline view/frontend/web/js/view/payment/method-renderer/paystandmagento-directpost.js:98]: space `//This`

File: view/frontend/web/js/view/payment/method-renderer/paystandmagento-directpost.js (hunk 1 above)
Comment (jmbeltran07) [reply view/frontend/web/js/view/payment/method-renderer/paystandmagento-directpost.js]: Done!

File: view/frontend/web/js/view/payment/method-renderer/paystandmagento-directpost.js (hunk 2 above)
Comment (jmbeltran07) [reply view/frontend/web/js/view/payment/method-renderer/paystandmagento-directpost.js]: Done

File: view/frontend/web/js/view/payment/method-renderer/paystandmagento-directpost.js (hunk 3 above)
Comment (jmbeltran07) [reply view/frontend/web/js/view/payment/method-renderer/paystandmagento-directpost.js]: done

File: view/frontend/web/js/view/payment/method-renderer/paystandmagento-directpost.js (hunk 4 above)
Comment (jmbeltran07) [reply view/frontend/web/js/view/payment/method-renderer/paystandmagento-directpost.js]: Done

**Comment by HFR1994** [comment]
QA approved

**Pull request merged**